### PR TITLE
feat(ereal): refactor division.cpp to standard test pattern (#947)

### LINKED ISSUE #947: Refactor elastic/ereal/arithmetic/division.cpp to standard test pattern
**Parent epic: #944**

## Goal

Bring `elastic/ereal/arithmetic/division.cpp` up to the standard regression-test structure.

## Division-specific test coverage

Division is NOT commutative or associative, but has its own invariants:

- Identity: `a / 1 == a`
- Self-quotient: `a / a == 1` (for finite non-zero `a`)
- Sign rules: `(-a) / b == -(a/b)`, `a / (-b) == -(a/b)`, `(-a) / (-b) == a/b`
- Exact divisions: `6 / 2 == 3` (no residual)
- Power-of-two divisors: `a / 2^k` should be exact for representable `a`
- Recurring quotients: `1 / 3` populates multiple limbs
- Division by zero handling: per `EREAL_THROW_ARITHMETIC_EXCEPTION` mode
- Subnormal boundary cases

## Constraint

Single-operator: tests exercise `/` / `/=` only.

## Pattern

See `addition.cpp` (PR #943).

## Acceptance

- Single-operator (`/` and `/=` only in the operation-under-test).
- Standardised framework.
- Builds clean gcc + clang, debug + release.

### PR DESCRIPTION
**Pull request by Ravenwater**
## Summary
- Refactors `elastic/ereal/arithmetic/division.cpp` to the standard regression-test structure (matching `addition.cpp` PR #943, `subtraction.cpp`, `multiplication.cpp`).
- Operation under test restricted to `/` / `/=`; `+` / `-` appear only in input construction.

Relates to #944 (parent epic). Resolves #947.

## Changes
- `elastic/ereal/arithmetic/division.cpp`:
  - anonymous-namespace `VerifyErealDivision(reportTestCases)` + property-based `VerifyErealDivision_Fuzz` (fixed seed, `(seed, iter)` reported on failure)
  - `REGRESSION_LEVEL_1..4` guards (x1k/x10k/x100k/x1M) + `ReportTestSuiteHeader`/`ReportTestResult`/`ReportTestSuiteResults`
  - removed the `(a/b)*b == a` identity (uses `*` -> belongs in `identities.cpp`), the non-ASCII `~=` chars, and the ad-hoc `main()`

## Coverage (per #947)
identity (`a/1`), self-quotient (`a/a==1`), sign rules (`(-a)/b`, `a/(-b)`, `(-a)/(-b)`), exact divisions (`6/2`, `1/4`), power-of-two divisor (exact exponent shift), recurring quotient (`1/3` multi-limb), subnormal numerator, divide-by-zero.

## Exact vs. tolerance
ereal division is `e * reciprocal(f)` with a 3-iteration Newton reciprocal, so it is **not** exact in general. Division by 1, by a power of two, by an exact-reciprocal divisor (`6/2`, `1/4`), and the sign rules are bit-exact via `operator==`; self-quotient, `15/3`, and `1/3` carry a Newton residual and are checked on the projected `double` within a relative tolerance.

## Divide-by-zero (deferred)
The foundational suite asserts only the **stable current behavior** (`finite/0` projects to a non-finite double). The IEEE-correct results (`finite/0` = signed Inf, `0/0` = NaN) and the `EREAL_THROW_ARITHMETIC_EXCEPTION` throwing path are non-IEEE / unwired today; that test block is gated behind `EREAL_TEST_ISSUE_968` and tracked as **#968** (the way #962 followed #945 and #966 followed #946).

## Test Results
| Target | gcc build | gcc test | clang build | clang test |
|--------|-----------|----------|-------------|------------|
| er_arith_division | OK | PASS (L1 + x1k + x10k) | OK | PASS (L1 + x1k + x10k) |

## Test plan
- [x] Fast CI passes (gcc + clang CI_LITE)
- [x] Promote to ready when satisfied: `gh pr ready <NNN>`

Resolves #947

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Replaced ad-hoc division test with a comprehensive regression suite covering identity, tolerant self-quotient, sign rules, in-place parity, exact divisions, multi-limb recurring quotients, subnormal preservation, and divide-by-zero boundary checks (optional IEEE-like behavior).
  * Added a relative-tolerance comparator, a fixed-seed fuzzing harness with configurable intensities, and integrated standardized test reporting with manual mode.

[Review Change Stack](https://app.coderabbit.ai/change-stack/stillwater-sc/universal/pull/969?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: elastic/ereal/arithmetic/division.cpp [hunk 1]
```diff
@@ -1,140 +1,386 @@
...
 // Copyright (C) 2017 Stillwater Supercomputing, Inc.
 // SPDX-License-Identifier: MIT
 //
 // This file is part of the universal numbers project, which is released under an MIT Open Source license.
 #include <universal/utility/directives.hpp>
+#include <cmath>
+#include <limits>
+#include <random>
 #include <universal/number/ereal/ereal.hpp>
 #include <universal/verification/test_suite.hpp>
```
[the remark this answers is not in the document]
Comment (Ravenwater) [reply elastic/ereal/arithmetic/division.cpp:40]: Fixed in 8594da04 -- added `#include <algorithm>` for `std::max`. Verified clean syntax on gcc-13 and clang-18.
